test/testdata/infer/isa_generic.rb

```diff
@@ -32,3 +32,11 @@ def f(x)
     T.assert_type!(x, Concrete)
   end
 end
+
+sig { params(x: T.any(Base, Other)).void }
+def f2(x)
+  if x.is_a?(Base)
+    T.assert_type!(x, Base)
```

Comment:
I'm a fan of using `T.reveal_type` for asserting types instead of `assert_type!` because it'll alert a future contributor when an asserted type becomes more / less specific but still satisfies the subtyping relationship that `assert_type!` is checking.

Reply:
That works for me. I was just trying to match the ambient style in this test.